packages/core/src/__tests__/ceramic.test.ts

```diff
@@ -140,7 +140,12 @@ describe('Ceramic integration', () => {
       await TestUtils.waitForState(
         stream3,
         2000,
-        (state) => StreamUtils.statesEqual(state, stream1.state),
+        (state) => {
+          // The timestamp deleted here will only be present if the log
```

Comment:
I don't quite understand why this is necessary.  The commits *are* being synced over the network, via pubsub UPDATE messages?

Reply:
Oh, is the issue that if we apply a signed commit, and then later on apply an anchor commit, we don't back-propagate the timestamp into the CommitDatas for earlier commits in the log?

If so, isn't that a problem for this whole approach?  If the stream state was updated one commit at a time instead of synced via the network all at once, then loading at a CommitID will fail again since the state log won't have the timestamp information where it's needed?  I think loading a tip and syncing the full log at once vs loading and applying commits one at a time as they come in through pubsub UPDATE messages, should both result in the same stream state.  If they don't, that's a bug IMO.  So we should probably make applying an anchor commit actively propagate the timestamp back to earlier entries in the log (until the previous anchorCommit)

Reply:
Yep, this is a good point.